src/algebra/module/graded_module.lean

```diff
@@ -0,0 +1,249 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import ring_theory.graded_algebra.basic
+import algebra.direct_sum.decomposition
+
+/-!
+# Graded Module
+
+Given an `R`-algebra `A` graded by `𝓐`, a graded `A`-module `M` is expressed as
+`direct_sum.decomposition 𝓜` and `set_like.has_graded_scalar 𝓐 𝓜`.
+Then `⨁ i, 𝓜 i` is an `A`-module and is isomorphic to `M`.
+
+## Tags
+
+graded module
+-/
+
+
+open_locale direct_sum big_operators
+
+variables {ι R A M σ : Type*}
+variables [add_monoid ι] [comm_semiring R] [semiring A] [algebra R A]
+variables (𝓐 : ι → submodule R A)
```

Comment:
Now that `graded_ring` is merged, you should be able to change this to
```suggestion
variables (𝓐 : ι → σ') [set_like σ' A] [add_submonoid_class σ' A]
```
or similar

Reply:
done